walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,84 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) { 
+    // check edge cases before calling helper functions
+    if (request.getDuration() == 0 || request.getDuration() > 1440 ) { 
```

Comment:
What does `1440` represent? For constants it's useful to pull them out with a descriptive variable name, I.e. `const int MINUTES_IN_A_DAY = 1440;`